src/models/position.jsx

```diff
@@ -0,0 +1,52 @@
+import { createContext, useMemo } from "react";
+
+const PositionContext = createContext({
+  getCoordinate: () => {},
+});
+
+const BingKey = "AgXLKJqvvaCBQb9-OrR7_pON_UPvsZafaJEPU9LRkXBREwvv3Z6jvilQG1EexTuA";
+
+const PositionContextProvider = ({ children }) => {
+  const getCoordinate = async (street, city, state, zipcode) => {
+    const countryRegion = "US";
+    try {
+      const response = await fetch(
+        "http://dev.virtualearth.net/REST/v1/Locations/" +
+          countryRegion +
+          "/" +
+          state +
+          "/" +
+          zipcode +
+          "/" +
+          city +
+          "/" +
+          street +
+          "?key=" +
+          BingKey
+      );
+      const json = await response.json();
+      const latitude = json.resourceSets[0].resources[0].point.coordinates[0];
+      const longitude = json.resourceSets[0].resources[0].point.coordinates[1];
+      return {
+        latitude: latitude,
+        longitude: longitude,
+      };
+    } catch (error) {
+      console.error("Error fetching location:", error);
```

Comment:
the current website logic is the functional part only get data when promise resolves, and  throw error (not handling it) if rejects.

all the error handling is left to the displaying component, it will either display error in the website or do an error toast.

so you should only throw error here, not consuming the error

a thing I will not request to change:
this can be written only as a function, not a context.